File: apps/api/app/schemas/investor_discovery.py
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, Field


class InvestorDiscoveryItem(BaseModel):
    id: UUID
    user_id: UUID
    name: str
    organization: str
    investment_thesis: str
    ticket_size: Decimal
    verification_badges: list[str] = Field(default_factory=list)

    model_config = {"from_attributes": True}


class InvestorDiscoveryResponse(BaseModel):
    items: list[InvestorDiscoveryItem]
    total: int
    limit: int
    offset: int

File: apps/web/src/lib/api/investor-discovery.ts
import { apiClient } from "@/lib/api/client";
import type { VerificationBadge } from "@/lib/api/profile-verification";

export type InvestorDiscoveryItem = {
  id: string;
  user_id: string;
  name: string;
  organization: string;
  investment_thesis: string;
  ticket_size: string;
  verification_badges: VerificationBadge[];
};

export type InvestorDiscoveryResponse = {
  items: InvestorDiscoveryItem[];
  total: number;
  limit: number;
  offset: number;
};

export type InvestorDiscoveryParams = {
  query?: string;
  organization?: string;
  ticket_min?: string;
  ticket_max?: string;
  limit?: number;
  offset?: number;
};

export async function discoverInvestors(params: InvestorDiscoveryParams) {
  const response = await apiClient.get<InvestorDiscoveryResponse>("/investor-discovery", {
    params
  });

  return response.data;
}
